npm/builder.js

```diff
@@ -487,6 +490,17 @@ Builder.prototype.compile = function() {
   fs.createReadStream(asciidoctorCSSFile).pipe(fs.createWriteStream('build/asciidoctor.css'));
 };
 
+Builder.prototype.replaceUnsupportedFeatures = function() {
+  log.title('Replace unsupported features');
+  var path = 'build/asciidoctor-core.js';
+  var data = fs.readFileSync(path, 'utf8');
+  log.debug('Replace sub! with sub');
+  data = data.replace(/\['\$sub!'\]/gi, '[\'$sub\']');
```

Comment:
So the regexp needs to be a bit more complex in order to capture the return value.

The following pattern:

```
($d = ($e = result)['$gsub!']
```

must become

```
result = ($d = ($e = result)['$gsub!']
```

which gives us:

```js
data.replace(/(\(\$\w+ = \(\$\w+ = (\w+)\))\['(\$g?sub)!'\]/g, "$2 = $1['$3']")
```